### PR TITLE
[DEV-2727] enable multi-address swap

### PR DESCRIPTION
**Pull request by lapatric**

### CODE REVIEW

File: src/screens/swap.screen.tsx [hunk 1]
```diff
@@ -263,7 +272,12 @@ export default function SwapScreen(): JSX.Element {
     return () => {
       isRunning = false;
     };
-  }, [validatedData]);
+  }, [validatedData, session?.address]);
+
+  async function onSwitchUser(address: string): Promise<void> {
+    const { accessToken } = await changeUserAddress(address);
+    setSession(accessToken);
+  }
```
Comment (davidleomay) [inline src/screens/swap.screen.tsx:280]: Do not overwrite the access token

**Review by lapatric — COMMENTED** [review]
- [ ] Update packages version for new paymentInfo fetching logic

File: src/screens/swap.screen.tsx [hunk 2]
```diff
@@ -115,6 +117,10 @@ export default function SwapScreen(): JSX.Element {
     sourceAssets && getBalances(sourceAssets).then(setBalances);
   }, [getBalances, sourceAssets]);
 
+  useEffect(() => {
+    console.log('receiveFor changed');
+  }, [receiveFor]);
```
Comment (davidleomay) [inline src/screens/swap.screen.tsx:122]: Remove log

File: src/screens/swap.screen.tsx [hunk 3]
```diff
@@ -144,15 +150,16 @@ export default function SwapScreen(): JSX.Element {
...
       ? [
-          ...targetBlockchains.map((b) => ({
-            address: blankedAddress(session.address ?? ''),
-            label: toString(b),
-            chain: b,
-          })),
+          ...userAddresses.flatMap((a) =>
+            targetBlockchains.map((b) => ({ address: a ?? '', label: toString(b), chain: b })),
+          ),
           {
```
Comment (davidleomay) [inline src/screens/swap.screen.tsx:163]: Every linked address has it's own list of available blockchains

**Pull request merged**